plugins/modules/net_tools/snmp_facts.py

```diff
@@ -298,9 +297,7 @@ def main():
         module.fail_json(msg=missing_required_lib('pysnmp'), exception=PYSNMP_IMP_ERR)
 
     cmdGen = cmdgen.CommandGenerator()
-    transport_opts = {'retries': m_args['retries']}
-    if m_args['timeout']:
-        transport_opts['timeout'] = m_args['timeout']
+    transport_opts = dict((k, m_args[k]) for k in ('timeout', 'retries') if m_args[k])
```

Comment:
```suggestion
    transport_opts = dict((k, m_args[k]) for k in ('timeout', 'retries') if m_args[k] is not None)
```
Since 0 is a valid value (at least for `retries`).